### PR TITLE
Refactor variable names and update doc

### PR DESCRIPTION
**Pull request by zaro0508**
* For clarification, change principal_id to synapse_owner_id
* Update docs with dependencies info

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +16,20 @@ utility is used to deploy the macro that invokes the lambda as a CloudFormation
...
+AWS will apply the following tags when resources are provisioned with the Service Catalog:
+
+|Key                                        |Value (something like)                                                    |
+|-------------------------------------------|--------------------------------------------------------------------------|
+|aws:servicecatalog:provisioningPrincipalArn|arn:aws:sts::123456712:assumed-role/ServiceCatalogEndusers/1234567        |
+|aws:servicecatalog:provisionedProductArn   |arn:aws:servicecatalog:us-east-1:123456712:stack/my-product/pp-mycpuogt2i45s|
+
+This custom resource uses these tags to retrieve more information and applies
+them as additional tags on the provisioned resource.
+
```
Comment (tthyer) [inline README.md:32]: This is a nice addition

File: set_tags/set_bucket_tags.py [hunk 2]
```diff
@@ -54,8 +54,8 @@ def create_or_update(event, context):
   log.info('Start Lambda processing')
   bucket_name = get_bucket_name(event)
   bucket_tags = get_bucket_tags(bucket_name)
-  principal_id = utils.get_principal_id(bucket_tags)
-  synapse_tags = utils.get_synapse_tags(principal_id)
+  synapse_ower_id = utils.get_synapse_ower_id(bucket_tags)
```
Comment (tthyer) [inline set_tags/set_bucket_tags.py:57]: ower? owner?

File: set_tags/set_instance_tags.py [hunk 3]
```diff
@@ -76,8 +76,8 @@ def create_or_update(event, context):
   log.info('Start Lambda processing')
   instance_id = get_instance_id(event)
   instance_tags = get_instance_tags(instance_id)
-  principal_id = utils.get_principal_id(instance_tags)
-  synapse_tags = utils.get_synapse_tags(principal_id)
+  synapse_owner_id = utils.get_synapse_ower_id(instance_tags)
```
Comment (tthyer) [inline set_tags/set_instance_tags.py:79]: please correct ower -> owner

File: set_tags/utils.py [hunk 4]
```diff
@@ -28,7 +28,7 @@ def get_iam_client():
 def get_synapse_client():
   return synapseclient.Synapse()
 
-def get_principal_id(tags):
+def get_synapse_ower_id(tags):
```
Comment (tthyer) [inline set_tags/utils.py:31]: fix "ower"

File: tests/unit/utils/test_get_synapse_owner_id.py [hunk 5]
```diff
@@ -3,15 +3,15 @@
 from set_tags import utils
 
 
-class TestGetPrincipalId(unittest.TestCase):
+class TestGetSynapseOwnerlId(unittest.TestCase):
```
Comment (tthyer) [inline tests/unit/utils/test_get_synapse_owner_id.py:6]: TestGetSynapseOwnerlId -> TestGetSynapseOwnerId

**Review by tthyer — CHANGES_REQUESTED** [review]

File: set_tags/set_bucket_tags.py (hunk 2 above)
Comment (zaro0508) [reply set_tags/set_bucket_tags.py]: fixed

File: set_tags/set_instance_tags.py (hunk 3 above)
Comment (zaro0508) [reply set_tags/set_instance_tags.py]: fixed

File: set_tags/utils.py (hunk 4 above)
Comment (zaro0508) [reply set_tags/utils.py]: fixed

File: tests/unit/utils/test_get_synapse_owner_id.py (hunk 5 above)
Comment (zaro0508) [reply tests/unit/utils/test_get_synapse_owner_id.py]: fixed

**Review by tthyer — APPROVED** [review]

**Pull request merged**